n_queen_problem/NQueenProblem.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Implementation of popular N-Queen Problem using Backtracking algorithm.
+ * The size of a chess board is given, suppose n, then we have to suggest such a configuration than n queens could be placed on the chess board and no queen is under attack
+ * There can be a number of possible solutions for a specific board.
+ * This implementation prints only one valid configuration, it can be extended to print all possible valid configurations.
+ * A good example of recursion.
+ * You can see the solution for various sizes by changing the value of Size in 14th line..
+ */
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+using namespace std;
+
+#define side 8                          /* Size of the board = (side X side) */
```

Comment:
As `side` is a constant, an all caps name will suit it better. This will help it differentiate from common variables which have lowercase names. Don't you agree?